syntax = "proto2";

package interconnection;

message command_type {
  enum command_t {
    PING = 0;
    PONG = 1;
    DRONE_COORDINATES = 2;
    MISSION_START = 3;
    MISSION_PAUSE = 4;
    MISSION_CONTINUE = 5;
    MISSION_ABORT = 6;
    LASER_RANGE = 7;
  }

  required command_t type = 1;
  required int32 version = 2;
}

message packet_size {
  required fixed32 size = 1;
}

message drone_coordinates {
  enum state_t {
    READY = 0;
    WAITING = 1;
    PAUSED = 2;
    EXECUTING = 3;
  }

  required double latitude = 1;
  required double longitude = 2;
  required float heading = 3;
  required state_t state = 4;
  required int32 event_id = 5;
}

message event_id_message {
  required int32 event_id = 1;
}

message pin_coordinates {
  required double latitude = 1;
  required double longitude = 2;
  required int32 event_id = 3;
}

message laser_range {
  required float range = 1;
}
